### PR TITLE
fix(ui): stabilize active-run steering queue

### PR DESCRIPTION
**Pull request by cryppadotta**
## Thinking Path

> - Paperclip is the open source app people use to manage AI agents for work.
> - The issue detail page shows a live agent run and accepts follow-up instructions.
> - A follow-up must stay in a stable queue until the user sends, reorders, or removes it.
> - Native runners can receive a steering event in the active run.
> - Legacy runners must interrupt the active run and start a follow-up run.
> - The current UI moved comments between the queue and the transcript and could show duplicate text or ambiguous chronology.
> - This pull request makes the queue projection durable, keeps each message in one clear place, and labels when queued input was actually steered or delivered.
> - The benefit is predictable steering with stable ordering, no duplicate messages, and visible causal timing.

## Linked Issues or Issue Description

Refs #11374.
Refs #12591.

**What happened?**

During an active run, a new follow-up could first appear as a transcript bubble and then move into the steering queue. After a steer or remove action, it could appear again. Progress text could also repeat the final response text. Once consumed, a queued bubble displayed only its original submission time even though it moved to its later causal slot, and a native run split by steering looked like two unrelated runs.

**Expected behavior**

An active-run follow-up must appear in the queue immediately. A native steer must move it once into the active run. A legacy interrupt must move it once into the follow-up run. A removed item must stay removed. Progress text that is identical to the final response must appear once. Consumed follow-ups must show both queue and steer/delivery times, and post-steer native segments must identify themselves as continuations of the same run.

**Steps to reproduce**

1. Start a long-running task.
2. Send two or more follow-up messages while the agent is active.
3. Reorder the messages and remove one message.
4. Send the first queued message as steering.
5. Observe the queue and transcript during and after both runs.

**Paperclip version or commit**

The problem reproduced on commit `da1e40302`.

**Deployment mode**

Local development with the embedded database.

## What Changed

- Project queued comments into the steering well for native and legacy live runners.
- Send native steering to the active run and use interrupt-and-follow-up for legacy runners.
- Keep optimistic queue order stable across refreshes and roll back failed actions.
- Remove discarded comments from the transcript cache and keep them removed when the queue becomes empty.
- Collapse only the final progress occurrence matching the durable response, including across steered transcript segments.
- Show `Queued … · Steered …` for same-run input and `Queued … · Delivered …` for successor-run input at their causal positions.
- Label settled and live post-steer segments `Continued after steering` and time them from the steer boundary.
- Add regression tests for queue display, steering, fallback interrupt, reorder, remove, rollback, duplicate text, causal timestamps, and live/settled continuation headers.

## Verification

- Ran the final focused steering/chronology UI suite with 233 passing tests.
- Ran the activity-service regression suite with 5 passing tests.
- Ran the broader queue-focused UI suite with 298 passing tests before the final chronology refinement.
- Ran `pnpm -r typecheck` successfully.
- Ran `pnpm build` successfully.
- Ran `pnpm check:token-gates` successfully.
- Tested native steering in a real browser with a 90-second baseline wait and a three-second steering correction.
- Confirmed that the old final response did not appear before the steered response.
- Tested three queued messages in a real browser.
- Confirmed that reorder changed delivery order and that the removed message was never sent or shown again.
- Tested a legacy runner in a real browser.
- Confirmed that it used the interrupt fallback and showed the follow-up once.
- Reloaded a saved mixed-steer/successor-run thread and confirmed the causal timestamps and continuation header render in the correct positions.
- The complete macOS suite reaches five unrelated platform assertions in workspace-runtime tests. Two compare `/var` with `/private/var`. Three require Linux `/proc` listener data. GitHub Actions provides the authoritative Linux run.

## Risks

- Low risk. The change is limited to issue-chat queue projection and transcript presentation.
- The server run-history API adds only a read-only `contextIssueId` projection; the database schema does not change.
- Optimistic actions restore the prior UI state when a request fails.

## Model Used

- OpenAI Codex with GPT-5, extended reasoning, browser automation, shell tools, and code execution.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have searched GitHub for duplicate or related PRs and linked them above
- [x] I have either (a) linked existing issues with `Fixes: #` / `Closes #` / `Refs #` OR (b) described the issue in-PR following the relevant issue template
- [x] I have not referenced internal/instance-local Paperclip issues or links (only public GitHub `#NNN` / `github.com/paperclipai/paperclip` URLs)
- [x] My branch name describes the change (e.g. `docs/...`, `fix/...`) and contains no internal Paperclip ticket id or instance-derived details
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] I have updated relevant documentation to reflect my changes
- [x] I have considered and documented any risks above
- [x] All Paperclip CI gates are green
- [x] Greptile is 5/5 with no open P2s, recommendations, or follow-ups
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

**Comment by cryppadotta** [comment]
Addressed the Greptile finding in 0da520fbd: settled native-run deduplication now scans all steering segments together and removes only the final matching progress occurrence. Added both a helper-level regression and a TaskChatThread integration regression proving an intentional pre-steer repeat remains visible while the redundant post-steer copy is suppressed. Focused tests (159), UI typecheck, full workspace typecheck/build, and token gates pass locally.

File: ui/src/components/task-chat/task-chat-adapter.ts [hunk 1]
```diff
@@ -46,6 +46,32 @@ export function formatTaskChatTimestamp(value: unknown): string | undefined {
...
+): string | undefined {
+  const queuedAt = formatTaskChatTimestamp(comment.createdAt);
+  const deliveredAt = formatTaskChatTimestamp(comment.conversationAnchorAt);
+  const isDeliveredFollowUp = Boolean(
+    kind === "human" &&
+    comment.conversationAnchorAt &&
+    comment.consumedByRunId &&
+    (comment.followUpRequested ||
+      comment.steeredIntoRunId ||
+      (queuedAt && deliveredAt && queuedAt !== deliveredAt)),
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/components/task-chat/task-chat-adapter.ts]: Fixed in 1991f31fd. The timestamp adapter now relies only on explicit follow-up or steering provenance. For older rows that lack the flag, IssueDetail recovers it from the stronger historical fact that the comment was submitted during a different run execution interval. Added regressions proving delayed ordinary run-start comments keep a compact timestamp and historical queued follow-ups retain Queued/Delivered labeling. Focused suite: 231 passing; full typecheck, build, and token gates pass locally.

File: ui/src/pages/IssueDetail.tsx [hunk 2]
```diff
@@ -1705,6 +1726,31 @@ const IssueDetailChatTab = memo(function IssueDetailChatTab({
...
+          const finishedAtMs = run.finishedAt
+            ? new Date(run.finishedAt).getTime()
+            : Number.POSITIVE_INFINITY;
+          return (
+            Number.isFinite(startedAtMs) &&
+            Number.isFinite(finishedAtMs) &&
+            startedAtMs <= submittedAtMs &&
+            submittedAtMs <= finishedAtMs
+          );
+        }),
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/pages/IssueDetail.tsx]: Fixed in 9535b9d4a. Historical provenance now requires the comment to have been submitted during the target run’s immediately preceding completed run for the same agent. An unrelated overlapping linked run is no longer sufficient. Added an IssueDetail regression covering a normal run started while another activity-linked run overlaps; it remains unlabelled as a follow-up. The existing historical successor-delivery regression still passes.

File: ui/src/pages/IssueDetail.tsx [hunk 3]
```diff
@@ -1705,6 +1726,56 @@ const IssueDetailChatTab = memo(function IssueDetailChatTab({
...
+                  Number.isFinite(finishedAtMs) &&
+                  finishedAtMs <= targetStartedAtMs
+                );
+              })
+              .sort(
+                (a, b) =>
+                  new Date(b.finishedAt!).getTime() -
+                  new Date(a.finishedAt!).getTime(),
+              )[0]
+          : undefined;
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply ui/src/pages/IssueDetail.tsx]: Fixed in 6d03de4ad. The run ledger now exposes contextIssueId from the durable run context. Legacy provenance recovery accepts any completed same-agent source run whose contextIssueId matches the issue and whose execution contains the submission, before the consuming target run; activity-only linked runs are excluded. Added regressions for both an unrelated overlap false positive and an unrelated intervening run between the genuine source and successor. Focused UI tests: 233 passed; activity service: 5 passed; UI/server typechecks passed.
